### DISCUSSION TITLE
After switching from app insights to Aspire's opinionated OTEL defaults, x-forwarded-for no longer works

### DISCUSSION THREAD

**Discussion by onionhammer** [microsoft/aspire#3006] [General]
After removing app insights telemetry and replacing it with Aspire's OOTB OTEL hookup (plus the azure log exporter) now every single request is "from" whatever ACA region the app is hosted in (in this case Des Moines, Iowa).

Is there some guidance on how to actually configure OTEL to use the X-Forwarded-For header and set the exported value in azure logs?

**Comment by davidfowl** [comment]
I think the place to file these issues is https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter.  Have you tried updating the package version? I think there were some changes made to make ACA work better.

**Reply by onionhammer** [reply]
@davidfowl are you referring to the beta? We're currently using 1.2.0

**Reply by davidfowl** [reply]
Yes try the beta and see if that fixes it. Then we'll know if there's more work left to do (for the azure team to do 😄)

**Reply by onionhammer** [reply]
@davidfowl I just confirmed that 1.3.0-beta.1 doesnt fix it

**Reply by davidfowl** [reply]
OK please open an issue on that repo and I'll get some eyes to look at it.

**Reply by onionhammer** [reply]
@davidfowl logged https://github.com/Azure/azure-sdk-for-net/issues/42850

**Comment by davidfowl** [comment]
See https://github.com/dotnet/aspire/issues/2047

**Comment by samsp-msft** [comment]
Reading the description in https://github.com/Azure/azure-sdk-for-net/issues/42850, this is not really Aspire related - its about differences between the old dedicated ApplicationInsights SDK and the OTel Exporter.

**Reply by davidfowl** [reply]
It's not but can you follow up? It affects aspire apps.
